go-controller/pkg/network-controller-manager/node_network_controller_manager.go

```diff
@@ -0,0 +1,223 @@
+	eventRecorder record.EventRecorder) *nodeNetworkControllerManager {
+	ncm := &nodeNetworkControllerManager{
+		name:         name,
+		client:       ovnClient.KubeClient,
+		Kube:         &kube.Kube{KClient: ovnClient.KubeClient},
+		watchFactory: wf,
+		stopChan:     make(chan struct{}),
+		recorder:     eventRecorder,
+	}
+
+	// dpu host mode does not need to do anything special for secondary network support
+	if config.OVNKubernetesFeature.EnableMultiNetwork && config.OvnKubeNode.Mode != ovntypes.NodeModeDPUHost {
+		klog.Infof("Multiple network supported, creating %s", controllerName)
+		ncm.nadController = newNetAttachDefinitionController(ncm, ovnClient.NetworkAttchDefClient, eventRecorder)
+	}
+	return ncm
+}
+
+// getOVNIfUpCheckMode check if OVN PortBinding.up can be used
+func (ncm *nodeNetworkControllerManager) getOVNIfUpCheckMode() error {
+	// this support is only used when configure Pod's OVS interface, it is not needed in DPU host mode
+	if config.OvnKubeNode.DisableOVNIfaceIdVer || config.OvnKubeNode.Mode == ovntypes.NodeModeDPUHost {
+		klog.Infof("'iface-id-ver' is manually disabled, ovn-installed feature can't be used")
+		ncm.isOvnUpEnabled = false
+		return nil
+	}
+
+	isOvnUpEnabled, err := util.GetOVNIfUpCheckMode()
+	if err != nil {
+		return err
+	}
+	ncm.isOvnUpEnabled = isOvnUpEnabled
+	if isOvnUpEnabled {
+		klog.Infof("Detected support for port binding with external IDs")
+	}
+	return nil
+}
+
+// Init initializes the node network controller manager and create default controller
+func (ncm *nodeNetworkControllerManager) Init() error {
+	if err := ncm.getOVNIfUpCheckMode(); err != nil {
+		return err
+	}
+
+	// Initialize OVS exec runner; find OVS binaries that the CNI code uses.
+	// Must happen before calling any OVS exec from pkg/cni to prevent races.
+	// Not required in DPUHost mode as OVS is not present there.
+	if err := cni.SetExec(kexec.New()); err != nil {
+		return err
+	}
+
+	ncm.defaultNodeNetworkController = node.NewDefaultNodeNetworkController(ncm.newCommonNetworkControllerInfo())
+	return nil
+}
+
+// Start initializes and starts the node network controller manager, which handles both default and secondary controllers
+func (ncm *nodeNetworkControllerManager) Start(ctx context.Context) error {
+	klog.Infof("OVN Kube Node initialization, Mode: %s", config.OvnKubeNode.Mode)
+
+	// Start the watch factory to begin listening for events
+	err := ncm.Init()
+	if err != nil {
+		return err
+	}
+
+	return ncm.Run(ctx)
+}
+
+// Run starts the node network controller manager, including default network controllers and the NAD controller
+// that handles all net-attach-def and the associated secondary network controllers.
+func (ncm *nodeNetworkControllerManager) Run(ctx context.Context) error {
+	err := ncm.watchFactory.Start()
+	if err != nil {
+		return err
+	}
+
+	if ncm.defaultNodeNetworkController != nil {
```

Comment:
not possible. I believe I added that check (in the previous changes for ovnkube-master NetworkController) because in downstream, we will do not have defaultNetworkController, as default network will be handled the same way as secondary networks (handled by NAD controller).

to avoid confusion, I'll just delete this check.